src/Block/MediaBlockService.php

```diff
@@ -92,7 +93,7 @@ public function configureEditForm(FormMapper $form, BlockInterface $block): void
                     'label' => 'form.label_class',
                     'required' => false,
                 ]],
-                [$this->getMediaBuilder(), null, []],
+                [$this->getMediaBuilder($form), null, []],
```

Comment:
You need to add 
```
if (!$form instanceof AdminFormMapper) {
            throw new \InvalidArgumentException('Media Block requires to be used in the Admin context');
        }
```
        
Like https://github.com/sonata-project/SonataPageBundle/blob/e690dbf137d5e30deb9a7a7dccbef1ffc627a3f5/src/Block/SharedBlockBlockService.php#L81-L85